src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,250 @@
+module sonic-pbh {
+
+	yang-version 1.1;
+
+	namespace "http://github.com/Azure/sonic-pbh";
+	prefix pbh;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-port {
+		prefix port;
+	}
+
+	import sonic-portchannel {
+		prefix lag;
+	}
+
+	description "PBH YANG Module for SONiC OS";
+
+	revision 2021-04-23 {
+		description "First Revision";
+	}
+
+	typedef hash-field {
+		description "Represents native hash field";
+		type enumeration {
+			enum INNER_IP_PROTOCOL;
+			enum INNER_L4_DST_PORT;
+			enum INNER_L4_SRC_PORT;
+			enum INNER_DST_IPV4;
+			enum INNER_SRC_IPV4;
+			enum INNER_DST_IPV6;
+			enum INNER_SRC_IPV6;
+		}
+	}
+
+	typedef packet-action {
+		description "Represents packet action";
+		type enumeration {
+			enum SET_ECMP_HASH;
+			enum SET_LAG_HASH;
+		}
+	}
+
+	typedef flow-counter {
+		description "Represents flow counter state";
+		type enumeration {
+			enum DISABLED;
+			enum ENABLED;
+		}
+	}
+
+	container sonic-pbh {
+
+		container PBH_HASH_FIELD {
+
+			description "PBH_HASH_FIELD part of config_db.json";
+
+			list PBH_HASH_FIELD_LIST {
+
+				key "hash_field_name";
+
+				leaf hash_field_name {
+					description "The name of this hash field";
+					type string;
+				}
+
+				leaf hash_field {
+					description "Configures native hash field for this hash field";
+					mandatory true;
+					type pbh:hash-field;
+				}
+
+				leaf ip_mask {
+					description "Configures IPv4/IPv6 address mask for this hash field";
+					mandatory true;
+					type union {
+						type inet:ipv4-address;
+						type inet:ipv6-address;
+					}
+					when "boolean(../hash_field[.='INNER_DST_IPV4' or .='INNER_SRC_IPV4' or .='INNER_DST_IPV6' or .='INNER_SRC_IPV6'])";
```

Comment:
Plz add a test case for this when condition.

Reply:
@praveen-li with the `union` type, there will be no way to have additional validation per `hash_field`, e.g. `ip_mask` can be `IPv4` and `hash_field` is `INNER_DST_IPV6` which will be invalid configuration. Not sure if we can handle it from `xpath` perspective.